src/backend/distributed/executor/executor.c

```diff
@@ -637,6 +664,21 @@ AssignTasksToConnections(DistributedExecution *execution)
 			connection = GetPlacementListConnectionIfCached(connectionFlags,
 															placementAccessList,
 															NULL);
+
+			/*
+			 * If there is no cached connection that we have to use and we're instructed
+			 * to force max query parallelization, we do it here with a blocking fashion,
+			 * contrary to the usual approach of non-blocking connection establishment.
+			 *
+			 * In the end, this is the user's choice and we have to open all the necessary
+			 * connections. Doing it here with blocking API simplifies the code a lot so we
+			 * prefer to do it.
+			 */
+			if (connection == NULL && ForceMaxQueryParallelization)
+			{
+				connection = GetPlacementListConnection(0, placementAccessList, NULL);
```

Comment:
Make it SESSION_LIFESPAN as done in the other place for completeness.

Reply:
`GetPlacementListConnection` synchronously establishes the connection, we should change this to `StartPlacementListConnection`